var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py

```diff
@@ -46,75 +46,79 @@ class EcpDataVisSdk(BundlePackage):
     variant('visit', default=False, description="Enable VisIt")
     conflicts('+visit')
 
-    ############################################################
-    # This is a messy workaround until the clingo concretizer can be required.
-    # The intent is to map package variants to dependency variants:
-    #   Package variants a, and b, mapping to dependency variants A and B
-    #   produce the following set of dependencies:
-    #     depends_on('foo+A+B', when='+a+b')
-    #     depends_on('foo+A~B', when='+a~b')
-    #     depends_on('foo~A+B', when='~a+b')
-    #     depends_on('foo~A~B', when='~a~b')
-    #   The clingo concretizer will allow that to be expressed much simpler by
-    #   only considering defaults once everything else is resolved:
-    #     depends_on('foo')
-    #     depends_on('foo+A', when='+a')
-    #     depends_on('foo+B', when='+b')
-    ############################################################
-
-    # Helper function to generate dependencies on the Cartesian product of
-    # variants.  If a dictionary is passed then it provides a mapping of
-    # package variant name to dependency variant name.  Otherwise assume they
-    # are the same variant name in both the package and dependency
-    def variants2deps(dep_spec, pkg_spec, variants):
-        if not type(variants) is dict:
-            variants = dict([(v, v) for v in variants])
-        n = len(variants)
-        for i in range(0, pow(2, n)):
-            state = ['+' if d == '1' else '~' for d in format(i, '0' + str(n) + 'b')]
-            [pkg_vars, dep_vars] = [''.join(v) for v in zip(
-                *[(s + pv, s + dv) for s, (pv, dv) in zip(state, variants.items())])]
-            dependency = ' '.join((dep_spec, dep_vars))
-            predicate = ' '.join((pkg_spec, pkg_vars))
-            depends_on(dependency, when=predicate)
+    # Wrapper around depends_on to propagate dependency variants
+    def dav_sdk_depends_on(spec, when=None, propagate=None):
```

Comment:
Can you use the `Spec` object to break down the dependency spec here instead of trying to do it manually?